code/modules/antagonists/changeling/powers/mutations.dm

```diff
@@ -463,6 +463,61 @@
 	if(ismob(loc))
 		loc.visible_message("<span class='warning'>[loc.name]\'s flesh turns black, quickly transforming into a hard, chitinous mass!</span>", "<span class='warning'>We harden our flesh, creating a suit of armor!</span>", "<span class='warning'>You hear organic matter ripping and tearing!</span>")
 
+
+/obj/item/clothing/suit/armor/changeling/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text, final_block_chance, damage, attack_type)
+	. = ..()
+
+	if(!IS_HORIZONTAL(owner))
+		return
+
+	var/obj/item/I = hitby
+	if(!istype(I))
+		return
+	if(!istype(I.loc, /mob/living))
+		return
+
+	var/mob/living/user = I.loc
```

Comment:
```suggestion
```
hit_reaction should never be called without passing a human mob, there's no point in istyping I.loc or checking who is wearing the armor as it's already given to you in the `owner` param. I would just forgo using `user` at all here and default to `owner` as it's more specifically typed.

Reply:
`I.loc` is actually checking the person who's holding the weapon that's being used to carve away the mass, not the person under the armor. Unless I'm completely mistaken, I think this makes sense as-is.

Reply:
👍 that makes sense, my mistake